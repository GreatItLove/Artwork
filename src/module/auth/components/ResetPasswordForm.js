import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';

import InputField from '../../common/components/form/InputField';
import SubmitButton from '../containers/SubmitButton';
import { LOGIN_PATH } from '../../../config/routesPatch';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  resetPassword: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};

/**
 * @class ResetPasswordForm
 * @extends React.Component
 * @description Render component
 */
class ResetPasswordForm extends Component {
  handleSubmitForm = data => {
    const { newPassword, uuid } = data;
    this.props.resetPassword({ password: newPassword, uuid });
  };

  /**
   * @description render
   * @return {Object} JSX HTML Content
   */
  render() {
    const { handleSubmit, t } = this.props;
    return (
      <div className="auth-login-wrapper">
        <div className="auth-login-fields">
          <div className="py-2" />
          <h4>{t('passwordReset.formHeader')}</h4>
          <div className="py-3" />
          <form onSubmit={handleSubmit(this.handleSubmitForm)}>
            <span>{t('passwordReset.headerInfo')}</span>
            <Field
              id="new_password"
              name="newPassword"
              component={InputField}
              floatingLabelText={t('passwordReset.newPassword.field')}
              type="password"
            />
            <Field
              id="confirm_password"
              name="confirmPassword"
              component={InputField}
              floatingLabelText={t('passwordReset.confirmPassword.field')}
              type="password"
            />
            <div className="py-2" />
            <div className="pt20">
              <SubmitButton label={t('passwordReset.submitBtn')} />
            </div>
            <div className="pt20 links">
              <Link to={LOGIN_PATH}>{t('passwordForget.backToLogin')}</Link>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

ResetPasswordForm.propTypes = propTypes;

export default ResetPasswordForm;
