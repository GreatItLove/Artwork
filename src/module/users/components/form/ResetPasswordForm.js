import React from 'react';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';

import Panel from '../../../common/components/Panel';
import SystemMessage from '../../../notifications/system/components/SystemMessage';
import PasswordField from '../../../common/components/form/PasswordField';
import ConfirmPasswordField from '../../../common/components/form/ConfirmPasswordField';
import CheckBox from '../../../common/components/form/CheckBox';
import SubmitUserFormButton from '../../containers/SubmitUserFormButton';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  submitForm: PropTypes.func.isRequired,
  closeDialog: PropTypes.func.isRequired,
  initialValues: PropTypes.shape({
    userId: PropTypes.number.isRequired
  }).isRequired
};

/**
 * @function ResetPassword
 * @param {Function} handleSubmit
 * @param {Function} submitForm
 * @param {Function} closeDialog
 * @returns {XML}
 */
function ResetPassword({ handleSubmit, submitForm, closeDialog }) {
  return (
    <Panel title="Reset password" panelHeadingClassName="users-from-panel-heading">
      <SystemMessage />
      <form onSubmit={handleSubmit(submitForm)} className="readmin-form inline users-form">
        <div className="row">
          <div className="col-md-6">
            <PasswordField floatingLabelText="Password" />
          </div>
          <div className="col-md-6">
            <ConfirmPasswordField floatingLabelText="Confirm Password" />
          </div>
          <div className="col-md-12 mt-4 mb-4">
            <Field
              name="forcePasswordChange"
              component={CheckBox}
              label="Require a change of password in the next sign in"
            />
          </div>
        </div>
        <div className="row mt-3 mb-3">
          <div className="col-md-6" />
          <div className="col-md-2 my-auto">
            <span className="cancel-link" onClick={closeDialog}>
              Cancel
            </span>
          </div>
          <div className="col-md-4">
            <SubmitUserFormButton label="Reset Password" />
          </div>
        </div>
      </form>
    </Panel>
  );
}

ResetPassword.propTypes = propTypes;

export default ResetPassword;
