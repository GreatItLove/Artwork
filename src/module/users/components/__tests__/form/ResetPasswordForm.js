import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ResetPasswordForm from '../../form/ResetPasswordForm';

describe('ResetPasswordForm', () => {
  it('Snapshot of ResetPasswordForm', () => {
    const props = {
      handleSubmit: jest.fn(),
      submitForm: jest.fn(),
      closeDialog: jest.fn(),
      initialValues: {
        userId: 4646
      }
    };

    const wrapper = shallow(<ResetPasswordForm {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
