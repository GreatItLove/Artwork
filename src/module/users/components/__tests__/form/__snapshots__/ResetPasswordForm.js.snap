// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm Snapshot of ResetPasswordForm 1`] = `
<Component
  panelHeadingClassName="users-from-panel-heading"
  title="Reset password"
>
  <Connect(SystemMessage) />
  <form
    className="readmin-form inline users-form"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <PasswordField
          floatingLabelText="Password"
        />
      </div>
      <div
        className="col-md-6"
      >
        <ConfirmPasswordField
          floatingLabelText="Confirm Password"
        />
      </div>
      <div
        className="col-md-12 mt-4 mb-4"
      >
        <Field
          component={[Function]}
          label="Require a change of password in the next sign in"
          name="forcePasswordChange"
        />
      </div>
    </div>
    <div
      className="row mt-3 mb-3"
    >
      <div
        className="col-md-6"
      />
      <div
        className="col-md-2 my-auto"
      >
        <span
          className="cancel-link"
          onClick={[Function]}
        >
          Cancel
        </span>
      </div>
      <div
        className="col-md-4"
      >
        <Connect(MuiComponent)
          label="Reset Password"
        />
      </div>
    </div>
  </form>
</Component>
`;
